addons/account/models/res_partner_bank.py

```diff
@@ -215,3 +281,16 @@ def unlink(self):
             msg = _("Bank Account %s with number %s deleted", account._get_html_link(title=f"#{account.id}"), account.acc_number)
             account.partner_id._message_log(body=msg)
         return super().unlink()
+
+    def name_get(self):
+        res = super().name_get()
+        if 'display_account_trust' in self.env.context and self.env.context['display_account_trust']:
+            res = []
+            for acc in self:
+                trusted_label = 'trusted' if acc.allow_out_payment else 'untrusted'
+                if acc.bank_id:
+                    name = '{} - {} ({})'.format(acc.acc_number, acc.bank_id.name, trusted_label)
+                else:
+                    name = '{} ({})'.format(acc.acc_number, trusted_label)
```

Comment:
```suggestion
                if acc.bank_id:
                    name = '{} - {} ({})'.format(acc.sanitized_acc_number, acc.bank_id.name, trusted_label)
                else:
                    name = '{} ({})'.format(acc.sanitized_acc_number, trusted_label)
```

Reply:
In the base name_get it is using acc_number, changing this isn't asked by the spec so I prefer to not do it